src/index.ts

```diff
@@ -53,5 +53,6 @@ export {
   ProcessPayload,
   DistinctivePropertiesPayload,
   BotDetectionResult,
-  BotKind,
+  BotKindType as BotKind,
```

Comment:
@xnerhu this type will produce string union of enum values as you suggested [here ](https://github.com/fingerprintjs/BotD/pull/122#issuecomment-1528823679)

Values are exported as separate token `BotKindEnum`

Feel free to suggest other approaches

Reply:
Is it necessary to change in this PR? Since it works, I'd not change unless necessary. But this is just my opinion. Both options are ok to me.

Reply:
BTW this is a minor but breaking change. Before `BotKind` could be used as a runtime value, not it can't.

Reply:
Yes, it's definitely a breaking change, I tried to re-export this enum using internal `const enum` but it also violates `isolatedModule` rule 

I think this approach is the most convenient

Reply:
See the difference between these two enums (first one is with numeric values and the other with string values):
<details>
<summary>Screenshot</summary>
<img width="670" alt="image" src="https://github.com/fingerprintjs/BotD/assets/21976058/2d3529e5-f947-4a4e-8076-52b96e8d8405">
</details>

I suppose we had issues only with numeric enums, but there is a chance that something will go wrong when we change compiler settings (e.g. rollup plugins)

So I think it's safer to not to use runtime enums in this case, especially considering that they produc more verbose code after compilation

Reply:
The string enum looks prone to the same problem (IIFE with no return value). It is not used by Pro agent directly or indirectly, so I think it's ok to leave it as it was.

> So I think it's safer to not to use runtime enums in this case

What do you suggest: An object, a const enum or something else?

Reply:
> What do you suggest: An object, a const enum or something else?

I mean to keep object enum + type as separate tokens

Reply:
What about the breaking change?

Reply:
I think we could respect that in version change

Or I can try to rewrite internal implementation to `const enum` (to keep bundle small) but use separate enum  or public API (perhaps simple test will help to keep it in sync)

I can see implicit issue that we might meet in the future if we will use this enum indirectly:

```
if (BotKind.Value) {
  return 0
} else {
  return -1
}
```

In this case everything will be valid from the API contract perspective (no empty values) but the logic is broken. 

But if you consider this as low risk I 100% don't mind to revert last commit to keep this PR simpler and avoid breaking changes

Reply:
> I think we could respect that in version change

What do you mean?

> Or I can try to rewrite internal implementation to `const enum` (to keep bundle small) but use separate enum or public API (perhaps simple test will help to keep it in sync)

Looks inconvenient to maintain. Do you suggest `const enum` to reduce the minified code size?

> I can see implicit issue that we might meet in the future if we will use this enum indirectly

I don't understand the issue. Could you please rephrase?

> But if you consider this as low risk I 100% don't mind to revert last commit to keep this PR simpler and avoid breaking changes

Let the maintainers decide. I understand the urge for consistency, but keeping a regular enum for `BotKind` seems to be the best option in this case.

Reply:
> What do you mean?

I mean to use major increment for example

> Looks inconvenient to maintain. Do you suggest const enum to reduce the minified code size?

Not only, I mean to keep this enum resistant to tree shaking (previous enums were used only indirectly, we didn't import them explicitly)

> I don't understand the issue. Could you please rephrase?

Sure, I mean these type of comparisons `BotKind.Value === 'value'` will silently become broken (because right now we only can track API contract in integration test but not the correct values

This is good a example of case when broken runtime enums are hard to notice:
https://github.com/fingerprintjs/BotD/blob/eac928d967c72032924de595a82850c0bf483005/src/detectors/window_size.ts#L5

`State.Success` will be equal to `undefined` and therefore the logic based on this condition will be affected, even if the API contract is valid (it will return `false` as expected in some cases)

> Let the maintainers decide. I understand the urge for consistency, but keeping a regular enum for BotKind seems to be the best option in this case

Ok, I agree

Reply:
> I mean to use major increment for example

Ok. Bumping the major version just for this change is too much, I think, more breaking changes should be collected.

> This is good a example

Now I understand, thanks.

@xnerhu Your decision is needed here.